sqlobject/src/main/java/org/jdbi/v3/sqlobject/internal/SqlObjectInitData.java

```diff
@@ -121,6 +102,139 @@ public Supplier<InContextInvoker> lazyInvoker(Object target, Method method, Hand
+     * @deprecated
+     */
+    @Deprecated
+    public static Method lookupMethod(String methodName, Class<?>... parameterTypes) {
+        return GeneratorSqlObjectFactory.lookupMethod(methodName, parameterTypes);
+    }
+
+    private static SqlObjectInitData initDataFor(ConfigRegistry handlersConfig, Class<?> sqlObjectType) {
+        Map<Method, Handler> methodHandlers = buildMethodHandlers(handlersConfig, sqlObjectType);
+
+        // process all annotations on the type.
+        final ConfigurerMethod forType = (configurer, config, annotation) -> configurer.configureForType(config, annotation, sqlObjectType);
+
+        // build a configurer for the type and all supertypes. This processes all annotations on classes and interfaces
+        UnaryOperator<ConfigRegistry> instanceConfigurer = buildConfigurers(
+                Stream.concat(JdbiClassUtils.superTypes(sqlObjectType), Stream.of(sqlObjectType)), forType);
+
+        // A map for all methods that contains the annotations for each method on the type
+        Map<Method, UnaryOperator<ConfigRegistry>> methodConfigurers = methodHandlers.keySet().stream().collect(Collectors.toMap(
+                        Function.identity(),
+                        method -> {
+                            // process all annotations on a single method.
+                            final ConfigurerMethod forMethod = (configurer, config, annotation) ->
+                                    configurer.configureForMethod(config, annotation, sqlObjectType, method);
+                            // build a configurer that processes all annotations on the method itself.
+                            return buildConfigurers(Stream.of(method), forMethod);
+                        }));
+
+        return new SqlObjectInitData(
+                sqlObjectType,
+                instanceConfigurer,
+                methodConfigurers,
+                methodHandlers);
+    }
+
+    private static Map<Method, Handler> buildMethodHandlers(
+            ConfigRegistry config,
+            Class<?> sqlObjectType) {
+
+        final Handlers handlers = config.get(Handlers.class);
+        final HandlerDecorators handlerDecorators = config.get(HandlerDecorators.class);
+        final Map<Method, Handler> handlerMap = new HashMap<>();
+
+        // tostring can not be constant, different for every object
+        Handler toStringHandler = (target, args, handleSupplier) ->
+                "Jdbi sqlobject proxy for " + sqlObjectType.getName() + "@" + Integer.toHexString(target.hashCode());
+
+        addMethodHandler(handlerMap, toStringHandler, Object.class, "toString");
+        addMethodHandler(handlerMap, EQUALS_HANDLER, Object.class, "equals", Object.class);
+        addMethodHandler(handlerMap, HASHCODE_HANDLER, Object.class, "hashCode");
+
+        // this should probably have a check whether the sqlObjectType actually implements SqlObject
```

Comment:
I don't think it is harmful to have extra handler methods here that you can't actually call.

Reply:
that comment has always been there.